### PR TITLE
temp branch syd_new exclude auto_generated directorys from ESI-SYD

### PR DESCRIPTION
**Pull request by yudongsi**
1.  add analyze_perf_logs.py file
2. add mltprocess in auto_gen.py
3. update models.py
4. update rewrite_models.py 
6. add parallel in test_generated_models.sh
7. add batchsize by modifying generate_file func in utils.py

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -58,6 +58,6 @@ def get_inputs_dict(name, inputs):
                 inputs_dict[inputs[i]] = "torch.ones((1, 384), dtype=torch.long)"
             for i in range(3, len(inputs)):
                 inputs_dict[inputs[i]] = "None"
-    elif name in vision_models or name in geffnet_models or name in segmentation_models:
```
Comment (blzheng) [inline models.py]: Segmentation models are supported now. Please revert this change.

File: rewrite_models.py [hunk 2]
```diff
@@ -47,15 +49,24 @@ def rewrite_model(filename, inputs_dict, module_dict, attr_dict, forward_list):
             f.write(opstr.replace(",)\n", ")\n"))
         f.write("\n")
         f.write("m = M().eval()\n")
-        # f.write("ref_m = "+pretrained_models_str[modelname]+".eval()\n")
+        f.write("CORES=os.popen(\"lscpu | grep Core | awk '{print $4}'\").readlines()\n")
+        f.write("SOCKETS=os.popen(\"lscpu | grep Socket | awk '{print $2}'\").readlines()\n")
+        f.write("BS=int(CORES[0])*int(SOCKETS[0])\n")
+        f.write("batch_size=BS\n")
```
Comment (blzheng) [inline rewrite_models.py]: BS is not always the number of cores. Other batch sizes are needed as well, e.g., 1, core_num/2, core_num, core_num*2, etc. Maybe we can add batch_size as a parameter and pass them from outside.

File: rewrite_models.py [hunk 3]
```diff
@@ -47,15 +49,24 @@ def rewrite_model(filename, inputs_dict, module_dict, attr_dict, forward_list):
...
+        f.write("    for i in range(10):\n")
+        f.write("        output = m("+inputstr+")\n")
+        f.write("    total_iter_time = time.time() - start_time\n")
+        f.write("    Throughput = batch_size * 10 / total_iter_time\n")
+        f.write("    file_current = os.path.basename(__file__)\n")
+        f.write("    print(file_current,',',BS,',',flag,',',Throughput)\n")
+
+        f.write("for flag in {False,True}:\n")
+        f.write("    torch._C._jit_set_texpr_fuser_enabled(flag)\n")
+        f.write("    print_throughput(flag)\n")
```
Comment (blzheng) [inline rewrite_models.py]: This benchmark is not only used for NNC. I think the generated models should be general so that we do not need to spend time analyzing models and generating codes every time we have new requirements. So such specific settings need to be decoupled from auto-gen scripts. What do you think about it?

File: rewrite_models.py [hunk 4]
```diff
@@ -119,9 +132,24 @@ def rewrite_model_temp(filename, inputs_dict, module_dict, attr_dict, forward_li
                 f.write(get_print_str(modelname, op))
         f.write("\n")
         f.write("m = M().eval()\n")
+        f.write("CORES=os.popen(\"lscpu | grep Core | awk '{print $4}'\").readlines()\n")
+        f.write("SOCKETS=os.popen(\"lscpu | grep Socket | awk '{print $2}'\").readlines()\n")
+        f.write("BS=int(CORES[0])*int(SOCKETS[0])\n")
+        f.write("batch_size=BS\n")
```
Comment (blzheng) [inline rewrite_models.py]: ditto.

File: rewrite_models.py [hunk 5]
```diff
@@ -119,9 +132,24 @@ def rewrite_model_temp(filename, inputs_dict, module_dict, attr_dict, forward_li
...
+        f.write("    for i in range(10):\n")
+        f.write("        output = m("+inputstr+")\n")
+        f.write("    total_iter_time = time.time() - start_time\n")
+        f.write("    Throughput = batch_size * 10 / total_iter_time\n")
+        f.write("    file_current = os.path.basename(__file__)\n")
+        f.write("    print(file_current,',',BS,',',flag,',',Throughput)\n")
+
+        f.write("for flag in {False,True}:\n")
+        f.write("    torch._C._jit_set_texpr_fuser_enabled(flag)\n")
+        f.write("    print_throughput(flag)\n")
```
Comment (blzheng) [inline rewrite_models.py]: ditto.

File: test_generated_models.sh [hunk 6]
```diff
@@ -1,44 +1,55 @@
 #!/bin/bash
-cur_dir=`pwd`
-mkdir -p ${cur_dir}/models/
-mkdir -p ${cur_dir}/temp/
-mkdir -p ${cur_dir}/shapes/
-mkdir -p ${cur_dir}/submodules/
+CORES=`lscpu | grep Core | awk '{print $4}'`
+SOCKETS=`lscpu | grep Socket | awk '{print $2}'`
+TOTAL_CORES=`expr $CORES \* $SOCKETS - 1`
+benchmark_dir="/home/pnp/yudongsi/benchmark"
```
Comment (blzheng) [inline test_generated_models.sh]: Please avoid using specific paths in the scripts. Suggest replacing it with an external environment variable.

File: utils.py [hunk 7]
```diff
@@ -171,8 +172,6 @@ def is_target(target, op):
         real = real.split(".")[-1]
     if re.match(target+'[0-9]*$', real):
         return True
-    if re.match(target+'[23]d[0-9]*$', real):
-        return True
```
Comment (blzheng) [inline utils.py]: Please revert this change. It fixed the bug that failed to identify patterns with the suffix "2d/3d".

File: utils.py [hunk 8]
```diff
@@ -287,9 +288,14 @@ def generate_file(filename, inputs, outputs, shapes_dict, module_dict, attr_dict
                 retstr = retstr + ", " + o
         f.write(retstr + "\n")
         f.write("\n")
-        f.write("m = M().eval()\n")
+        f.write("m = M().eval()\n\n")
+        f.write("CORES=os.popen(\"lscpu | grep Core | awk '{print $4}'\").readlines()\n")
+        f.write("SOCKETS=os.popen(\"lscpu | grep Socket | awk '{print $2}'\").readlines()\n")
+        f.write("BS=int(CORES[0])*int(SOCKETS[0])\n")
+        f.write("batch_size=BS\n")
```
Comment (blzheng) [inline utils.py]: ditto.

File: utils.py [hunk 9]
```diff
@@ -306,10 +312,19 @@ def generate_file(filename, inputs, outputs, shapes_dict, module_dict, attr_dict
...
+        f.write("    for i in range(10):\n")
+        f.write("        output = m("+inputstr+")\n")
+        f.write("    total_iter_time = time.time() - start_time\n")
+        f.write("    Throughput = batch_size * 10 / total_iter_time\n")
+        f.write("    file_current = os.path.basename(__file__)\n")
+        f.write("    print(file_current,',',BS,',',flag,',',Throughput)\n")
+
+        f.write("for flag in {False,True}:\n")
+        f.write("    torch._C._jit_set_texpr_fuser_enabled(flag)\n")
+        f.write("    print_throughput(flag)\n")
```
Comment (blzheng) [inline utils.py]: ditto.

File: test_generated_models.sh (hunk 6 above)
Comment (WeizhuoZhang-intel) [reply test_generated_models.sh]: Please do not remove those three lines for computing TOTAL_CORES, which will be used for the performance test in numactl. Not related to BS

File: models.py (hunk 1 above)
Comment (yudongsi) [reply models.py]: ok

File: rewrite_models.py (hunk 2 above)
Comment (yudongsi) [reply rewrite_models.py]: have revert it, done later.

File: rewrite_models.py (hunk 3 above)
Comment (yudongsi) [reply rewrite_models.py]: ok

File: test_generated_models.sh (hunk 6 above)
Comment (yudongsi) [reply test_generated_models.sh]: ok

File: utils.py (hunk 7 above)
Comment (yudongsi) [reply utils.py]: ok
